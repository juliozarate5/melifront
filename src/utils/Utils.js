export const truncateTitle = (text) => {
       return (text.length > 50) ? text.substring(0, 50) + "..." : text; 
}