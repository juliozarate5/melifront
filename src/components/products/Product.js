import React from 'react'
import { truncateTitle } from '../../utils/Utils';
import PropTypes from 'prop-types';

export default function Product({d, i}) {
    return (
        <div 
            className="col-lg-3 col-md-4 col-sm-2 col-12 mb-4"
            key={i}
        >
            <div 
                className="card border-0 shadow custom-card"
                
            >
                <a href={d.url} target="_blank">
                <img 
                    src={d.imageUrl} 
                    className="imagen" 
                    alt="Imagen"
                />
                </a>  
                <div className="card-body">
                    <p 
                        className="card-text"
                    >
                        {truncateTitle(d.title)}
                                                    
                    </p>
                </div>
            </div>
        </div>
    )
}

Product.propTypes = {
    d: PropTypes.object.isRequired,
    i: PropTypes.number.isRequired
}