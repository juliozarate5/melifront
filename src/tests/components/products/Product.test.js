import React from "react";
import { shallow } from "enzyme";
import Product from "../../../components/products/Product";
import {truncateTitle} from '../../../utils/Utils';
jest.mock('../../../utils/Utils');

describe('product tests', () => {

    const d = {};
    const i = 1;

    const wrapper = shallow(<Product d={d} i={i}/>)

    test('showing component successful', () => {

        expect(wrapper).toMatchSnapshot();
    });

});