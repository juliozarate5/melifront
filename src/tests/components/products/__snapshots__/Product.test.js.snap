// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product tests showing component successful 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-2 col-12 mb-4"
  key="1"
>
  <div
    className="card border-0 shadow custom-card"
  >
    <a
      target="_blank"
    >
      <img
        alt="Imagen"
        className="imagen"
      />
    </a>
    <div
      className="card-body"
    >
      <p
        className="card-text"
      />
    </div>
  </div>
</div>
`;
